polaris-react/src/components/TextField/TextField.tsx

```diff
@@ -656,16 +660,23 @@ export function TextField({
 
   function handleKeyPress(event: React.KeyboardEvent) {
     const {key, which} = event;
-    const numbersSpec = /[\d.eE+-]$/;
-    if (type !== 'number' || which === Key.Enter || numbersSpec.test(key)) {
+    const numbersSpec = /[\d.,eE+-]$/;
+    const integerSpec = /[\deE+-]$/;
```

Comment:
I don't think this prevents a user from copy + pasting a decimal number into the integer field. We may need to come up with a way to prevent this, e.g. filter the value before displaying.

Reply:
This would still allow non sensical input like `45-234234`, correct? Is there some way we could validate against that? 🤔 

Also, it looks like the [`keypress` event is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Element/keypress_event). Shouldn't we do the validation in `keydown` instead?

Reply:
Good spotting! Indeed we can do non-sensical input like you've described, although this isn't any different to the existing `number` type. 
I agree another layer of input validations would be useful, and we should be using `keydown` instead of `keypress`, but I don't have much context on this component and its architecture, so it's probably something someone from @Shopify/polaris-team needs to advise on.